jc/parsers/bluetoothctl.py

```diff
@@ -0,0 +1,366 @@
+    + r"|\s*Connected:\s*(?P<connected>.+)"
+    + r"|\s*LegacyPairing:\s*(?P<legacy_pairing>.+)"
+    + r"|\s*Modalias:\s*(?P<modalias>.+)"
+    + r"|\s*RSSI:\s*(?P<rssi>.+)"
+    + r"|\s*TxPower:\s*(?P<txpower>.+)"
+    + r"|\s*UUID:\s*(?P<uuid>.+))"
+)
+
+
+def _parse_device(next_lines: List[str]) -> Device:
+    next_line = next_lines.pop()
+    result = re.match(_device_head_pattern, next_line)
+
+    if not result:
+        next_lines.append(next_line)
+        return None
+    
+    matches = result.groupdict()
+
+    name = matches["name"]
+
+    if name.endswith("not available"):
+        return None
+
+    device = Device = {
+        "address": matches["address"],
+    }
+
+    if name.endswith("(public)"):
+        device["is_public"] = True
+        name = name.replace("(public)", "")
+    
+    device["name"] = name.strip()
+
+    while next_lines:
+        next_line = next_lines.pop()
+        result = re.match(_device_line_pattern, next_line)
+
+        if not result:
+            next_lines.append(next_line)
+            return device
+        
+        matches = result.groupdict()
+        
+        if matches["name"]:
+            device["name"] = matches["name"]
+        elif matches["alias"]:
+            device["alias"] = matches["alias"]
+        elif matches["class"]:
+            device["class"] = matches["class"]
+        elif matches["icon"]:
+            device["icon"] = matches["icon"]
+        elif matches["paired"]:
+            device["paired"] = matches["paired"]
+        elif matches["bonded"]:
+            device["bonded"] = matches["bonded"]
+        elif matches["trusted"]:
+            device["trusted"] = matches["trusted"]
+        elif matches["blocked"]:
+            device["blocked"] = matches["blocked"]
+        elif matches["connected"]:
+            device["connected"] = matches["connected"]
+        elif matches["legacy_pairing"]:
+            device["legacy_pairing"] = matches["legacy_pairing"]
+        elif matches["modalias"]:
+            device["modalias"] = matches["modalias"]
+        elif matches["rssi"]:
+            device["rssi"] = matches["rssi"]
+        elif matches["txpower"]:
+            device["txpower"] = matches["txpower"]
+        elif matches["uuid"]:
+            if not "uuids" in device:
+                device["uuids"] = []
+            device["uuids"].append(matches["uuid"])
+
+    return device
+
+def parse(data: str, raw: bool = False, quiet: bool = False) -> List[JSONDictType]:
+    """
+    Main text parsing function
+
+    Parameters:
+
+        data:        (string)  text data to parse
+        raw:         (boolean) unprocessed output if True
+        quiet:       (boolean) suppress warning messages if True
+
+    Returns:
+
+        List of Dictionaries. Raw or processed structured data.
+    """
+    jc.utils.compatibility(__name__, info.compatible, quiet)
+    jc.utils.input_type_check(data)
+
+    linedata = data.splitlines()
```

Comment:
Maybe move these two under the `if` statement? Not a big deal but no reason to process the data if the data check fails.